### PR TITLE
Introduce extended DApp API [#5042]

### LINKED ISSUE #5042: Introduce extended DApp API
### User Story

As a DApp developer, I want to have access to status specifics API in my DApp

### Description

A DApp will be able to access a status object allowing to access status specifics API. The syntax will follow the one defined in this [proposition](https://discuss.status.im/t/status-web-dapps-api/141).

The only API exposed at first will be `currentPublicKey`.

*Type*: Feature

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #5042

WIth this PR Dapp developers can request access to status web api. Only "CONTACT_CODE" permission implemented atm.

```js
window.addEventListener('message', function (event) {
    if (!event.data || !event.data.type) { return; }
    if (event.data.type === 'STATUS_API_SUCCESS') {
        console.log(event.data.permissions) //=> ["CONTACT_CODE"] if allowed , and [] if not allowed
        console.log(STATUS_API["CONTACT_CODE"]) //=> "0x0012300..123" if allowed
    }
});
// request status API
window.postMessage({ type: 'STATUS_API_REQUEST', permissions: ["CONTACT_CODE"] }, '*');
```

`STATUS_API` global object will be injected only if user allowed access

### CODE REVIEW

File: src/status_im/models/browser.cljs [hunk 1]
```diff
@@ -35,4 +38,36 @@
...
+(defn update-dapp-permissions-fx [{:keys [db]} permissions]
+  {:db            (assoc-in db [:dapps/permissions (:dapp permissions)] permissions)
+   :data-store/tx [(dapp-permissions/save-dapp-permissions permissions)]})
+
+(defn request-permission [cofx
+                          {:keys [dapp-name index requested-permissions permissions-allowed user-permissions
+                                  permissions-data webview]
+                           :as   params}]
+  (if (< index (count requested-permissions))
+    (let [requested-permission (get requested-permissions index)]
```
Comment (jeluard) [inline src/status_im/models/browser.cljs]: Can you comment what is going on here?

**Review by jeluard — COMMENTED** [review]
Please update the changelog! :)

**Comment by flexsurfer** [comment]
@jeluard , @rachelhamlin liked to update changelog :)

File: src/status_im/models/browser.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/models/browser.cljs]: a lot of interesting things happen here :)

Comment (flexsurfer) [reply src/status_im/models/browser.cljs]: comment here or in the code? i just iterate requested permissions

**Review by goranjovic — APPROVED** [review]

File: src/status_im/models/browser.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/models/browser.cljs]: In the code would be nice!

**Comment by flexsurfer** [comment]
PR can be tested in Status Test Dapp
image

**Comment by flexsurfer** [comment]
@jeluard done

**Comment by Serhy** [comment]
Tested this with Status simple Dapp on Android 6, 7, 8, iOS 11 and it looks good. Failed autotest is the known issue exists in GH already.

**Review by dmitryn — APPROVED** [review]

**Pull request merged**